src/object/schema_manager.c

```diff
@@ -15618,14 +15624,134 @@ sm_truncate_using_destroy_heap (MOP class_mop)
 }
 #endif
 
+/*
+ * sm_collect_truncatable_classes () - Collects OIDs of truncatable classes regarding the CASCADE option
+ *   return: NO_ERROR on success, non-zero for ERROR
+ *   class_mop(in):
+ *   trun_classes(in/out): a hash for skipping checked classes and collecting class OIDs
+ *   is_cascade(in): whether to cascade TRUNCATE to FK-referring classes
+ */
+int
+sm_collect_truncatable_classes (MOP class_mop, unordered_oid_set & trun_classes, bool is_cascade)
+{
+  int error = NO_ERROR;
+  SM_CLASS *class_ = NULL;
+  SM_CLASS_CONSTRAINT *pk_constraint = NULL;
+  SM_FOREIGN_KEY_INFO *fk_ref;
+  OID *fk_cls_oid;
+
+  error = au_fetch_class (class_mop, &class_, AU_FETCH_READ, DB_AUTH_SELECT);
+  if (error != NO_ERROR || class_ == NULL)
+    {
+      assert (er_errid () != NO_ERROR);
+      return er_errid ();
+    }
+
+  trun_classes.emplace (ws_oid (class_mop));
+
+  pk_constraint = classobj_find_cons_primary_key (class_->constraints);
+  if (pk_constraint == NULL)
```

Comment:
I think it is more clear to add the following condition '|| pk_constraint->fk_info == NULL'.

Reply:
Good idea. I've revised it: [5fc8943](https://github.com/CUBRID/cubrid/pull/2738/commits/5fc8943a059f0dcd1bb34790198e6334d89c5ca0)